### PR TITLE
adapt new im_shape in mask rcnn-fpn

### PR DESCRIPTION
**Pull request by yghstill**

### CODE REVIEW

File: ppdet/data/transform/operator.py [hunk 1]
```diff
@@ -177,62 +174,12 @@ def __init__(self, eigval, eigvec, alphastd=0.1):
...
         self.eigvec = np.array(eigvec).astype('float32')
 
-    def apply(self, sample, context=None):
+    def apply(self, sample):
         alpha = np.random.normal(scale=self.alphastd, size=(3, ))
         sample['image'] += np.dot(self.eigvec, self.eigval * alpha)
         return sample
 
 
-@register_op
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:186]: why removed

File: ppdet/data/transform/operator.py [hunk 2]
```diff
@@ -401,7 +348,7 @@ def apply(self, sample, context=None):
                 lambda img: cv2.cvtColor(self.apply_saturation(cv2.cvtColor(img, cv2.COLOR_RGB2HSV)), cv2.COLOR_HSV2RGB),
                 lambda img: cv2.cvtColor(self.apply_hue(cv2.cvtColor(img, cv2.COLOR_RGB2HSV)), cv2.COLOR_HSV2RGB),
             ]
-            distortions = np.random.permutation(functions)[:self.count]
+            distortions = np.random.permutation(functions)[:count]
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:351]: self.count?

File: ppdet/data/transform/operator.py [hunk 3]
```diff
@@ -678,15 +625,11 @@ def apply(self, sample, context=None):
...
-        if 'scale_factor' in sample:
-            scale_factor = sample['scale_factor']
-            sample['scale_factor'] = np.array(
-                [scale_factor[0] * im_scale_y, scale_factor[1] * im_scale_x],
-                dtype=np.float32)
-        else:
-            sample['scale_factor'] = np.array(
-                [im_scale_y, im_scale_x], dtype=np.float32)
+        sample['im_shape'] = np.asarray([resize_h, resize_w], dtype=np.float32)
+        scale_factor = sample['scale_factor']
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:629]: what about scale_factor not in sample

File: ppdet/data/transform/operator.py [hunk 4]
```diff
@@ -749,33 +692,33 @@ def __init__(self,
...
                 format(type(target_size)))
         self.target_size = target_size
 
-        if not isinstance(origin_target_size, Sequence):
+        if not isinstance(origin_target_size, list):
             raise TypeError(
-                "Type of origin_target_size is invalid. Must be List or Tuple, now is {}".
-                format(type(origin_target_size)))
+                "Type of target_size is invalid. Must be List, now is {}".
+                format(type(target_size)))
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:704]: origin_target_size

File: ppdet/data/transform/operator.py [hunk 5]
```diff
@@ -871,10 +814,7 @@ def apply(self, sample, context=None):
         x = np.random.randint(0, w - width)
         offsets, size = [x, y], [h, w]
 
-        pad = Pad(size,
-                  pad_mode=-1,
-                  offsets=offsets,
-                  fill_value=self.fill_value)
+        pad = Pad(size, pad_mode=-1, offsets=offsets)
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:817]: fill_value

File: ppdet/data/transform/operator.py [hunk 6]
```diff
@@ -1545,32 +1485,30 @@ def __call__(self, sample, context=None):
...
-                (gt_score1 * factor, gt_score2 * (1. - factor)), axis=0)
-            result['gt_score'] = gt_score
-        if 'is_crowd' in sample[0]:
-            is_crowd1 = sample[0]['is_crowd']
-            is_crowd2 = sample[1]['is_crowd']
-            is_crowd = np.concatenate((is_crowd1, is_crowd2), axis=0)
-            result['is_crowd'] = is_crowd
-
-        return result
+        gt_bbox1 = sample[0]['gt_bbox']
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:1489]: do not change here?

File: ppdet/data/transform/operator.py [hunk 7]
```diff
@@ -1638,26 +1576,26 @@ def apply(self, sample, context=None):
         bbox = sample['gt_bbox']
         gt_num = min(self.num_max_boxes, len(bbox))
         num_max = self.num_max_boxes
-        # fields = context['fields'] if context else []
+        fields = context['fields'] if context else []
```
Comment (jerrywgz) [inline ppdet/data/transform/operator.py:1579]: do not change here

File: ppdet/modeling/head/mask_head.py [hunk 8]
```diff
@@ -149,12 +149,14 @@ def forward_test(self,
...
         else:
-            im_info_expand = []
+            scale_factor_list = []
             for idx, num in enumerate(bbox_num):
                 for n in range(num):
-                    im_info_expand.append(im_info[idx, -1])
-            im_info_expand = paddle.concat(im_info_expand)
-            scaled_bbox = paddle.multiply(bbox[:, 2:], im_info_expand, axis=0)
+                    scale_factor_list.append(scale_factor[idx, 0])
+            scale_factor_list = paddle.cast(
```
Comment (jerrywgz) [inline ppdet/modeling/head/mask_head.py:156]: scale_factor is already float32

File: ppdet/modeling/layers.py [hunk 9]
```diff
@@ -407,13 +407,10 @@ def __init__(self,
     def __call__(self, yolo_head_out, anchors, im_shape, scale_factor=None):
         boxes_list = []
         scores_list = []
-        im_shape = paddle.cast(im_shape, 'float32')
         if scale_factor is not None:
             origin_shape = im_shape / scale_factor
         else:
             origin_shape = im_shape
-
-        origin_shape = paddle.cast(origin_shape, 'int32')
```
Comment (jerrywgz) [inline ppdet/modeling/layers.py:416]: cast to int32 is need

File: ppdet/modeling/post_process.py [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+import paddle.fluid as fluid
```
Comment (jerrywgz) [inline ppdet/modeling/post_process.py:2]: remove

**Review by wangxinxin08 — COMMENTED** [review]
operator.py建议checkout回去，这部分的代码不要修改

**Pull request closed without merging**